program-runtime/src/compute_budget.rs

```diff
@@ -127,44 +127,93 @@ impl ComputeBudget {
         message: &SanitizedMessage,
         requestable_heap_size: bool,
         default_units_per_instruction: bool,
+        prioritization_fee_type_change: bool,
     ) -> Result<u64, TransactionError> {
         let mut num_instructions = message.instructions().len();
-        let mut requested_additional_fee = 0;
         let mut requested_units = None;
+        let mut requested_heap_size = None;
+        let mut prioritization_fee = None;
 
-        let error = TransactionError::InstructionError(0, InstructionError::InvalidInstructionData);
         for (i, (program_id, instruction)) in message.program_instructions_iter().enumerate() {
             if compute_budget::check_id(program_id) {
                 // don't include request instructions in default max calc
                 num_instructions = num_instructions.saturating_sub(1);
 
-                // Compute budget instruction must be in the 1st 3 instructions (avoid
-                // nonce marker), otherwise ignored
-                if i < 3 {
+                if prioritization_fee_type_change {
+                    let invalid_instruction_data_error = TransactionError::InstructionError(
+                        i as u8,
+                        InstructionError::InvalidInstructionData,
+                    );
+                    let duplicate_instruction_error =
+                        TransactionError::DuplicateInstruction(i as u8);
+
                     match try_from_slice_unchecked(&instruction.data) {
-                        Ok(ComputeBudgetInstruction::RequestUnits {
+                        Ok(ComputeBudgetInstruction::RequestUnitsDeprecated {
                             units,
                             additional_fee,
                         }) => {
                             requested_units = Some(units as u64);
-                            requested_additional_fee = additional_fee as u64;
+                            prioritization_fee = Some(additional_fee as u64);
```

Comment:
I think we should check if `requested_units` and `prioritization_fee` are already set above instead of overwriting them

Reply:
Still want a check for `prioritization_fee.is_some()` as well right?